src/inmanta/deploy/scheduler.py

```diff
@@ -372,6 +382,107 @@ async def get_facts(self, resource: dict[str, object]) -> None:
+                discrepancy_map[rid] = [
+                    Discrepancy(
+                        expected=(
+                            f"Resource {rid} is present in the DB (model version {latest_version}). "
+                            "It is expected in the scheduler's resource_state map."
+                        ),
+                        actual=f"Resource {rid} is missing from the scheduler's resource_state map.",
+                    )
+                ]
+
+            # Resources only present in the scheduler but missing from the DB
+            only_in_scheduler = self._state.resource_state.keys() - resources_in_db.keys()
+            for rid in only_in_scheduler:
+                discrepancy_map[rid] = [
+                    Discrepancy(
+                        expected=(
+                            f"Resource {rid} is not present in the DB (model version {latest_version}). "
+                            "It shouldn't be in the scheduler's resource_state map."
+                        ),
+                        actual=f"Resource {rid} is present in the scheduler's resource_state map.",
+                    )
+                ]
+
+            # For resources in both the DB and the scheduler, check for discrepancies in state
+            for rid in resources_in_db.keys() & self._state.resource_state.keys():
+                resource_discrepancies: list[Discrepancy] = []
+
+                db_resource_details = resources_in_db[rid]
+                db_deploy_result, db_blocked_status, db_compliance_status = state_translation_table[db_resource_details.status]
+
+                scheduler_resource_state: ResourceState = self._state.resource_state[rid]
+                if db_deploy_result:
+                    if scheduler_resource_state.deployment_result != db_deploy_result:
+                        resource_discrepancies.append(
+                            Discrepancy(expected=db_deploy_result, actual=scheduler_resource_state.deployment_result)
+                        )
+                if db_blocked_status:
+                    if scheduler_resource_state.blocked != db_blocked_status:
+                        resource_discrepancies.append(
+                            Discrepancy(expected=db_blocked_status, actual=scheduler_resource_state.blocked)
+                        )
+                if db_compliance_status:
+                    if scheduler_resource_state.status != db_compliance_status:
+                        resource_discrepancies.append(
+                            Discrepancy(expected=db_compliance_status, actual=scheduler_resource_state.status)
+                        )
+
+                if resource_discrepancies:
+                    discrepancy_map[rid] = resource_discrepancies
+
+            return discrepancy_map
+
+        discrepancy_map = _build_discrepancy_map(resources_in_db)
+        return SchedulerStatusReport(resource_state=self._state.resource_state, discrepancies=discrepancy_map)
```

Comment:
We may want to pack the DB state in here as well.

Reply:
How much of it should we pack ? I plopped everything from the `ResourceDetails` in there but this is maybe overkill ?